packages/web/components/cards/hero-card.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { memo, useEffect, useState } from "react";
+import { useTranslation } from "react-multi-lang";
+
+import { EventName } from "~/config";
+import { useAmplitudeAnalytics } from "~/hooks";
+
+import { Icon } from "../assets";
+import { IconLink } from "./icon-link";
+
+export const HeroCard: React.FunctionComponent<{
+  title: string;
+  subtitle: string;
+  imageUrl: string;
+  fallbackImageUrl?: string;
+  label?: string;
+  githubUrl?: string;
+  twitterUrl?: string;
+  externalUrl: string;
+  mediumUrl?: string;
+}> = memo(
```

Comment:
Is it necessary for us to use memoization here? Sometimes, preemptive optimizations can end up being costlier than not optimizing at all.

Reply:
memoization is never necessary, but when you don't expect the props to change is generally considered a good use case for the `memo` optimization: https://react.dev/reference/react/memo

Reply:
I don't see where the React docs recommend using `memo` for all possible use cases. As with all performance optimization methods within React, it introduces a small overhead. In some cases, the benefits outweigh the cost. However, I don't believe `memo` will bring significant benefits in this situation.